### ISSUE TITLE
Strange Observer issue duplicating components with Turbolinks

### ISSUE THREAD

**Issue by petermumford** [uikit/uikit#2541]
**Version: 3.0.0-beta.18**

When I click between pages the component attributes are duplicating. It never happens on the first click but will always start to happen after the third or fourth time of switching between pages. I can confirm this is a conflict with Turbolinks for Rails as it never happens when I turn Turbolinks off. 

As you can see from the screenshots below where the search icon has duplicated as well as extra placeholder div after the sticky component.

![screen shot 2017-03-22 at 16 43 53](https://cloud.githubusercontent.com/assets/331905/24209736/09bcc954-0f1f-11e7-9506-742f4f4e8ee0.png)

![screen shot 2017-03-22 at 16 46 43](https://cloud.githubusercontent.com/assets/331905/24209805/29e29cfe-0f1f-11e7-8583-4c0179ce8de2.png)

![screen shot 2017-03-22 at 16 44 17](https://cloud.githubusercontent.com/assets/331905/24209737/0bd34484-0f1f-11e7-8160-74aa9b521639.png)

**Comment by AdanCruzHuerta** [comment]
Hi. Did you solve your problem?

**Comment by petermumford** [comment]
@AdanCruzHuerta for now I'm just using the SVG html directly in my views instead of using the component attribute, in my case `uk-search-icon`. I also did the same with the `uk-sticky` component attribute by using the div's and classes it creates directly in my HTML view.

**Comment by respire** [comment]
+1
currently, i use this code as workaround
```.js
document.addEventListener('turbolinks:before-cache', () => {
  // workaround for uikit 3.0.0-beta.22
  document.querySelectorAll('*[uk-icon] > svg').forEach(elem => elem.remove())
})
```
hope this issue can be fixed!

**Comment by JRice** [comment]
FTR, respire's solution didn't *quite* work for me. I did this (assumes you have jQuery):

```js
if(!SOME_NAMESPACE) {
  var SOME_NAMESPACE = {};

  SOME_NAMESPACE.teardown = function() {
    console.log("TEARDOWN");
    $(".uk-search-icon > svg").remove();
  }
}

$(document).on("turbolinks:before-cache", SOME_NAMESPACE.teardown);
```

**Comment by briankelleher** [comment]
Was this truly fixed?  SVG icons are duplicating with 3.0.0-beta.35 and Turbolinks 5 still.  I still had to add the patch code above.

My simple ruby partial:

```erb
<% if bool %>
    <span class="green" uk-icon="icon: check"></span>
<% else %>
    <span class="red" uk-icon="icon: close"></span>
<% end %>
```

Still generates:

```html
<span class="green uk-icon" uk-icon="icon: check">
    <svg width="20" height="20" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg" ratio="1" style="transform: matrix(1, 0, 0, 1, 0, 0);"> <polyline fill="none" stroke="#000" stroke-width="1.1" points="4,10 8,15 17,4"></polyline></svg>
    <svg width="20" height="20" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg" ratio="1"> <polyline fill="none" stroke="#000" stroke-width="1.1" points="4,10 8,15 17,4"></polyline></svg>
</span>
```

Also curious that one generates a slightly different markup.

**Comment by SaphiLC** [comment]
not fixed... the issue is the cache, if you disable cache on turbolinks the problem dissapears because it re-loads the content instead of serving the cached page and triggering again everything (showing icons, generating a new sticky container, etc...). An option to "trash" every change uikit made to the cached page with javascript would fix the issue, at least for turbolinks5 since we can trigger that clean up on the before-cache event.

for now my solution with turbolinks5 is this
```
document.addEventListener('turbolinks:load', function() {
	Turbolinks.clearCache();
});
```
no cache, but in my case is fine since the pages are cached in the server, still getting jumps on stuff with margin added with javascript tho (like grids) but 0 duplicates
